LED_Marquee.ino

```diff
@@ -16,15 +16,20 @@ void diagnosticLedOff() {
     digitalWrite(diagnosticLed, LOW);   // turn the LED on (HIGH is the voltage level)
 }
 
-volatile boolean beat = false;
+volatile float beat = 0.0f;
+volatile unsigned long timeOfLastBeat = 0u;
+#define FADE_MILLIS = 200u; // number of ms beat flash takes to fade out
+
 // http://www.gammon.com.au/interrupts
 ISR (PCINT1_vect) {
     // handle pin change interrupt for A0 to A5 here
     if (PINC & bit (0)) {  // if it was high
-        beat = true;
+        beat = 1.0f;
+        timeOfLastBeat = millis();
         diagnosticLedOn();
     } else {
-        beat = false;
+        unsigned long delta = millis() - timeOfLastBeat;
+        beat = constrain((float)delta / (float)FADE_MILLIS, 0.0f, 1.0f);
```

Comment:
Does this count as a significant computation? ISRs should do as little as possible.

Reply:
Is `marquee()` run inside an ISR? I did not think it was. At any rate, this is not a significant computation. Although, if this loop is running inside an ISR, then the call to `millis()` will always return the same value on each cycle of the loop, which would be problematic.